ra/ra.go

```diff
@@ -715,7 +720,17 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizationsCAA(
 		} else if authz.Expires.Before(now) {
 			badNames = append(badNames, name)
 		} else if authz.Expires.Before(caaRecheckTime) {
-			recheckNames = append(recheckNames, name)
+			// If the name with an expired authorization is for a wildcard identifier we need to:
+			// a) strip the `*.` prefix to check CAA for the base hostname without the
+			//    wildcard
+			// b) note that the recheck needs to be treated as a wildcard name for CAA
+			if strings.HasPrefix(authz.Identifier.Value, "*.") {
+				baseName := strings.TrimPrefix(authz.Identifier.Value, "*.")
+				recheckNames[baseName] = true
```

Comment:
Will this not violate 6844 in the case where a request is made for a certificate containing `example.com` and `*.example.com` with a `issue` record denying issuance and a `issuewild` record allowing it?

Reply:
@rolandshoemaker Can you expand? I'm not sure I see how. The name `example.com` would be processed normally, checking the `issue` CAA record and being denied. The name `*.example.com` would be processed as a check for `issuewild` for `example.com`, which would be found and allowed. The overall issuance would fail because one of the two names was unauthorized. Am I missing something?

Reply:
Ahh, I think I understand now. The one stomps the other when we really need to do two distinct rechecks, one for the base domain as a normal domain, one for the base domain as a wildcard domain.

Reply:
This was a good catch. Thanks @rolandshoemaker! In bc2a722 I reworked the way that the RA collects up required CAA rechecks so that overlapping `*.example.com`, `example.com` checks are handled appropriately. I also updated the `TestRecheckCAADates` unit test to test that this is handled appropriately with two CAA checks for "example.com" - one as a wildcard, and one as a non-wildcard.